### PR TITLE
Add support for person type

### PR DESCRIPTION
**Pull request by yyccbb**
## What?
* Update `Person` model to capture its type (stu/ta) #39 :
  * stu = student
  * ta = TA
* Update `add` command to include (optional) person type information #36 :
  * When person type information is omitted, the person is registered as a student by default.
* Update `edit` command to support the new `Person` model
* Update `JsonAdaptedPerson` #41 
* Update relevant test cases to support all the changes #40 

## Why?
**TrAcker**  needs to distinguish between a student and a TA,

### CODE REVIEW

File: src/test/java/seedu/address/logic/parser/AddCommandParserTest.java [hunk 1]
```diff
@@ -165,32 +166,36 @@ public void parse_compulsoryFieldMissing_failure() {
...
         // two invalid values, only first invalid value reported
-        assertParseFailure(parser, INVALID_NAME_DESC + PHONE_DESC_BOB + EMAIL_DESC_BOB + INVALID_ADDRESS_DESC,
+        assertParseFailure(parser,
+                TYPE_DESC_BOB + INVALID_NAME_DESC + PHONE_DESC_BOB + EMAIL_DESC_BOB + INVALID_ADDRESS_DESC,
                 Name.MESSAGE_CONSTRAINTS);
 
-        // non-empty preamble
-        assertParseFailure(parser, PREAMBLE_NON_EMPTY + NAME_DESC_BOB + PHONE_DESC_BOB + EMAIL_DESC_BOB
-                + ADDRESS_DESC_BOB + TAG_DESC_HUSBAND + TAG_DESC_FRIEND,
-                String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
```
Comment (yongkotaro) [inline src/test/java/seedu/address/logic/parser/AddCommandParserTest.java:194]: Maybe can still leave this test case on top of invalid PersonType.

**Review by yongkotaro — COMMENTED** [review]
Maybe can add back in the non-empty preamble test case in src/test/java/seedu/address/logic/parser/AddCommandParserTest.java.

File: src/test/java/seedu/address/logic/parser/AddCommandParserTest.java (hunk 1 above)
Comment (yyccbb) [reply src/test/java/seedu/address/logic/parser/AddCommandParserTest.java:194]: The preamble part of an add command is used for recording the person type information so this invalidate the test case for non-empty preamble.

Comment (yongkotaro) [reply src/test/java/seedu/address/logic/parser/AddCommandParserTest.java:194]: Ok thank you for the clarification.

**Review by wongkj12 — APPROVED** [review]
LGTM!

File: src/test/java/seedu/address/logic/parser/EditCommandParserTest.java [hunk 2]
```diff
@@ -134,6 +134,7 @@ public void parse_oneFieldSpecified_success() {
         // name
         Index targetIndex = INDEX_THIRD_PERSON;
         String userInput = targetIndex.getOneBased() + NAME_DESC_AMY;
+        System.out.println(userInput);
```
Comment (kaitinghh) [inline src/test/java/seedu/address/logic/parser/EditCommandParserTest.java:137]: Was this left from debugging? Should we remove?

**Review by kaitinghh — APPROVED** [review]
Overall LGTM

File: src/test/java/seedu/address/logic/parser/EditCommandParserTest.java (hunk 2 above)
Comment (yyccbb) [reply src/test/java/seedu/address/logic/parser/EditCommandParserTest.java:137]: I think so. oops.

**Pull request merged**